.github/workflows/build.yml

```diff
@@ -5,32 +5,44 @@ on: [push, pull_request]
 jobs:
   build:
 
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        os: [ubuntu-latest, windows-latest]
         python-version: [3.7, 3.8]
 
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+    - name: Set up Miniconda with Python version ${{ matrix.python-version }}
+      uses: goanpeca/setup-miniconda@v1
       with:
+        auto-update-conda: true
+        channels: conda-forge
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
+      shell: bash -l {0}
       run: |
-        sudo apt install libgsl-dev   # Needed for msprime < 1.0. Binary wheels include GSL for >= 1.0
-        python -m pip install --upgrade pip
+        conda install msprime
         pip install -r requirements.txt -r requirements-dev.txt
     - name: Run pre-commit
-      uses: pre-commit/action@v2.0.0
+      if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.7'
+      shell: bash -l {0}
+      run: |
+        pre-commit install
```

Comment:
So this is not using cache for pre commit anymore? ` FYI pre-commit/action` comes with cache built in

Reply:
Here's an example of a run where it failed: https://github.com/tomwhite/sgkit/runs/1121992680?check_suite_focus=true

```
##[error]Unable to locate executable file: pre-commit. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
```